shell/src/main/java/alluxio/cli/validation/UfsDirectoryValidationTask.java

```diff
@@ -30,27 +31,41 @@
    * Creates a new instance of {@link UfsDirectoryValidationTask}
    * for validating root under file system.
    *
-   * @param conf configuration
+   * @param path the UFS path
+   * @param conf the UFS configuration
    */
-  public UfsDirectoryValidationTask(AlluxioConfiguration conf) {
-    mUfs = UnderFileSystem.Factory.createForRoot(conf);
-    mPath = conf.get(PropertyKey.MASTER_MOUNT_TABLE_ROOT_UFS);
+  public UfsDirectoryValidationTask(String path, AlluxioConfiguration conf) {
+    mPath = path;
+    mUfs = UnderFileSystem.Factory.create(mPath, conf);
   }
 
   @Override
-  public TaskResult validate(Map<String, String> optionsMap) {
+  public String getName() {
+    return "ValidateUfsDir";
+  }
+
+  @Override
+  public ValidateUtils.TaskResult validate(Map<String, String> optionsMap) {
+    StringBuilder msg = new StringBuilder();
+    StringBuilder advice = new StringBuilder();
     try {
       UfsStatus[] listStatus = mUfs.listStatus(mPath);
       if (listStatus == null) {
-        System.err.format("Unable to list under file system path %s.%n", mPath);
-        return TaskResult.FAILED;
+        msg.append(String.format("Unable to list under file system path %s. ", mPath));
```

Comment:
Now if the namenode / nameservice is not accessible there will be an exception here. I'm trying to add another test using `hadoopClient.isHealthy()` in hadoop client, which should be more verbose and specific.